storage/remote/queue_manager.go

```diff
@@ -684,14 +683,13 @@ func (t *QueueManager) AppendExemplars(exemplars []record.RefExemplar) bool {
 outer:
 	for _, e := range exemplars {
 		if isSampleOld(currentTime, time.Duration(t.cfg.SampleAgeLimit), e.T) {
-			t.metrics.droppedExemplarsTotal.Inc()
-			level.Debug(t.logger).Log("msg", "Dropped exemplar due to age", "ref", e.Ref)
+			t.metrics.droppedExemplarsTotal.WithLabelValues(reasonTooOld).Inc()
 			continue
 		}
 		t.seriesMtx.Lock()
 		lbls, ok := t.seriesLabels[e.Ref]
 		if !ok {
-			t.metrics.droppedExemplarsTotal.Inc()
+			t.metrics.droppedExemplarsTotal.WithLabelValues(reasonDroppedSeries).Inc()
```

Comment:
now that we have the `reason` label this line needs to move later (and in all similar boxes) so we can set the reason as `reasonDroppedSeries` or `reasonUnintentionalDroppedSeries`, something like that:

```
if _, ok := t.droppedSeries[e.Ref]; !ok {
    level.Info(t.logger).Log("msg", "Dropped exemplar for series that was not explicitly dropped via relabelling", "ref", e.Ref)
}
```

Reply:
You were thinking of something along the lines of  https://github.com/prometheus/prometheus/pull/13002/commits/25ec6b685743d3c47091610a682c8a921f737a97?